doc/userdoc/styleguide/styleguide.rst

```diff
@@ -0,0 +1,566 @@
+   ``iaf_psc_alpha``. We can modify the parameter `V_m` and set the value to
+   `50.0`.
+
+Rendered as
+
+
+   Here we use the ``Create`` function to instantiate our model, in this case
+   ``iaf_psc_alpha``. We can modify the parameters `V_m` and set the value to
+   `50.0`.
+
+\**Boldface\**
+~~~~~~~~~~~~~~
+
+If you want to emphasize a word or phrase in text, you can use **boldface**.
+
+NumPy style docstrings
+~~~~~~~~~~~~~~~~~~~~~~
+
+In the PyNEST code, we follow the rules for NumPy style docstrings as
+`explained here <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
+
+
+reStructuredText markup
+-----------------------
+
+Let us take a brief look into the markup of reStructuredText. You can find
+a full cheatSheet `here <https://thomas-cokelaer.info/tutorials/sphinx/
+rest_syntax.html>`_.
+
+.. note::
+
+   reStructuredText is sensitive to indentation. A single newline should always
+   be between any text and headings, tables, reference labels, or directives.
+   If the output format seems incorrect, double check the spaces and newlines
+   of the text.
+
+Code and code snippets
+~~~~~~~~~~~~~~~~~~~~~~
+
+For writing Python code, you can use the double colon
+
+
+ \::
+
+Rendered as
+
+   ::
+
+       python code here
+
+       x = 1
+
+       for i in line:
+           x = i
+           print(thing.x)
+
+For other code styles use the code-block directive (you can also use this one
+for python)
+
+
+
+\   .. code-block:: cpp
+
+Rendered as
+
+    .. code-block:: cpp
+
+       int main()
+       {
+         cout << "Hello, World!";
+         return 0;
+       }
+
+
+For input/output examples use the following syntax
+
+::
+
+   >>> input begins with 3, right-angled brackets
+   output is directly below input without any brackets
+
+
+Directives
+~~~~~~~~~~~
+
+This is not a complete guide to the directives of reStructuredText. For more
+options see `Sphinx directives <https://www.sphinx-doc.org/en/master/usage/
+restructuredtext/directives.html>`_ or `Docutils section on directives
+<http://docutils.sourceforge.net/docs/ref/rst/directives.html>`_.
+
+Equations
+^^^^^^^^^
+
+For equations, use LaTeX markup:
+
+::
+
+    .. math::
+
+            f(x) = \int_{-\infty}^{\infty} \hat f(\xi)\ e^{2 \pi i x \xi}\,d\xi,
+
+Rendered as
+
+
+    .. math::
+
+            f(x) = \int_{-\infty}^{\infty} \hat f(\xi)\ e^{2 \pi i x \xi}\,d\xi,
+
+
+Admonitions
+^^^^^^^^^^^
+
+Use admonitions to draw readers attention to a particular point. Possible
+admonitions types include
+
+"see also", "attention", "caution", "danger", "error", "hint", "important",
+"note", "tip", "warning", "admonition"
+
+If you want a custom admonition use
+
+::
+
+   .. admonition:: custom name
+
+         Here is some text
+
+Rendered as
+
+
+   .. admonition:: custom name
+
+         Here is some text
+
+
+Roles
+~~~~~
+
+See `the roles section in the Sphinx docs <https://www.sphinx-doc.org/en/
```

Comment:
I wonder if the split between roles and directives is not too technical, especially if it has not been explained before. As a doc writer, I want to put in math, probably mixing inline and displayed math, and then I would look for "how to put in math" and would expect information on inline and displayed math in one place.

I also find the term "Role" in RST-lingo very confusing.

Reply:
Good point, I'll update this.